web-common/src/features/dashboards/stores/dashboard-store-defaults.ts

```diff
@@ -42,11 +43,34 @@ export function setDefaultTimeRange(
   // TODO: refactor all sub methods and call setSelectedScrubRange here
   metricsExplorer.selectedScrubRange = undefined;
   metricsExplorer.lastDefinedScrubRange = undefined;
-  setDefaultComparisonTimeRange(metricsView, metricsExplorer, fullTimeRange);
+}
+
+function setDefaultComparison(
+  metricsView: V1MetricsViewSpec,
+  metricsExplorer: MetricsExplorerEntity,
+  fullTimeRange: V1ColumnTimeRangeResponse | undefined
+) {
+  switch (metricsView.defaultComparisonMode) {
+    case MetricsViewSpecDefaultComparisonMode.DEFAULT_COMPARISON_MODE_DIMENSION:
+      metricsExplorer.selectedComparisonDimension =
+        metricsView.defaultComparisonDimension ??
+        metricsView.dimensions?.[0]?.name;
```

Comment:
If not `dimension` key is provided along with mode "dimension", no comparison is shown.
This is because the value of `metricsView.defaultComparisonDimension` is an empty string in such cases.

Reply:
Weird. I have a test for this that passes.